go/chat/storage/storage.go

```diff
@@ -540,6 +546,9 @@ func (s *Storage) updateAllSupersededBy(ctx context.Context, convID chat1.Conver
 
 	// queue asset deletions in the background
 	s.assetDeleter.DeleteAssets(ctx, uid, convID, allAssets)
+	// queue search index update in the background
+	go s.G().Indexer.Remove(ctx, convID, uid, allPurged)
```

Comment:
Can we put the goroutine behavior inside this guy? I like to return a channel as a callback from these kinds of function to give the caller the option to wait or not.